packages/core/src/interpreter.ts

```diff
@@ -559,6 +590,14 @@ export class Interpreter<
       return this.state;
     }
 
+    if (((event as EventObject)?.type || '').includes('error')) {
```

Comment:
If anything - this should be checked on the `_event` (and on the appropriate field). It might be "too soon" to check this as the `event` might not be an `EventObject`, it actually might be already a `SCXML.Event`.

This should also check if the type/name of the event starts with `error.` and not if the type contains an `error` substring. 

also, a nit: optional chaining here and defaulting this to an empty string seems unnecessary:
- the `event` parameter is required
- all events should have valid types/names

Reply:
thank you, I updated it